### PR TITLE
[prometheus-to-sd] Improve source validation

### PR DESCRIPTION
**Pull request by Szetty**
Add tests for source validation

### CODE REVIEW

**Comment by Szetty** [comment]
cc @loburm

File: prometheus-to-sd/config/dynamic_source.go [hunk 1]
```diff
@@ -47,6 +47,9 @@ func validateSources(sources flags.Uris) (map[string]url.URL, error) {
 		if source.Key == "" {
 			return nil, errors.New("component name should NOT be empty for any dynamic source")
 		}
+		if source.Val.Port() == "" {
+			return nil, errors.New("port should NOT be empty for any dynamic source")
```
Comment (loburm) [inline prometheus-to-sd/config/dynamic_source.go]: Why do we need a second check?

https://github.com/GoogleCloudPlatform/k8s-stackdriver/blob/master/prometheus-to-sd/config/source_config.go#L50

Comment (Szetty) [reply prometheus-to-sd/config/dynamic_source.go]: to have the validation of all the required fields in one place, we should not always rely on the fact that the port will be validated after, also it is good to fail fast

Comment (Szetty) [reply prometheus-to-sd/config/dynamic_source.go]: also the port seems to be validated for static resources before too
https://github.com/GoogleCloudPlatform/k8s-stackdriver/blob/985653a490cab36c496aa666f410a0696816ee61/prometheus-to-sd/config/source_config.go#L79

**Pull request merged**